pyccel/ast/c_concepts.py

```diff
@@ -7,29 +7,52 @@
 Module representing object address.
 """
 
-from pyccel.utilities.metaclasses import Singleton
+from pyccel.utilities.metaclasses import ArgumentSingleton
 from .basic     import TypedAstNode, PyccelAstNode
-from .datatypes import DataType
+from .datatypes import HomogeneousContainerType, FixedSizeType, PyccelIntegerType
 from .literals  import LiteralString
 
 __all__ = ('CMacro',
+           'CNativeInt',
            'CStackArray',
            'CStringExpression',
            'ObjectAddress',
            'PointerCast')
 
 #------------------------------------------------------------------------------
 
-class CStackArray(DataType, metaclass=Singleton):
+class CNativeInt(FixedSizeType):
```

Comment:
what was the equivalent for representing C's native integer before and why is the class needed now?

Reply:
The equivalent for representing C's native integer in the `devel` branch is to use `precision = -2`. This was not a very good solution as native C integers are only used rarely so it is difficult for devs to remember what this precision corresponds to, however it was the only solution which matched the old type specification